aiohttp/client.py

```diff
@@ -226,9 +226,11 @@ def request(self, method, url, **kwargs):
 
         traces = [
             Trace(
-                trace_config,
                 self,
-                trace_request_ctx=trace_request_ctx)
+                trace_config,
+                trace_config.trace_config_ctx(
```

Comment:
Why trace config context is separated from trace_config object?

Reply:
There is a `trace_config_ctx` object instantiated per request, each handler receives as a param this instance to save information during the signals triggered by the same request.

The TraceConfig instance is registered as a param for the `ClientSession`, therefore the relationship between the config and the context is 1:N where N means requests.

Reply:
What if we put trace config inside context once context is created? Seems like they are bounded, so better not pass them around separated. 

Is there any cases when we could change context / context in the middle of request processing?

Reply:
IMHO I would prefer to keep the `trace_config` as an explicit argument of the signal handler. Both `trace_config` and `trace_config_ctx` have different purposes. Meanwhile, the first one is the class that allows you configure the signals that would be triggered, the second one is a pure temporary storage used to make the trace of a specific request stateful.

Change in the meaning of creating a new instance? No, I don't think so. Take into account that the pattern that we are allowing having this parameter is the following one

```python
ascync def start(session, trace_config, trace_config_ctx, *args, **kwargs):
    tace_config_ctx.start = session.loop.time()

ascync def end(session, trace_config, trace_config_ctx. *args, **kwargs):
    print("Total time {}".format(session.loop.time() - trace_config_ctx.start)
```

Reply:
thoughts ?

Reply:
Well, for me they looks like fragments of a single thing: `trace_config`.  Assume we have `TraceConfig` object. It has some attributes, methods, properties...well, everything about to configure the tracer as the name stands. At the same time it may wraps a block of code into the context, where some different rules shifts the original state, but only for that block.

I didn't dive into design deep enough to judje, but it just looks like that we can have a single `trace_config` object with `ctx` property or have standard context manager protocol with a bit help of `__call__` act in the similar way. But again, just a thoughts. And all because naming tells us that these two things are heavily coupled - probably too much to think that they could be the single entity.

Reply:
Anyway, this discussion is irrelevant to PR. It should be started on initial tracer PR, but there were too much changes to stop on every bit.